demo/AngryBots/Release/UnityLoader.js

```diff
@@ -158,25 +164,49 @@ var CompressionState = {
         }
     }
 };
+
+// For simplicity, .wasm files are not currently compressed but in a production
+// build, they would be (and use LoadCompressedFile).
+(function() {
+    var xhr = new XMLHttpRequest();
+    xhr.open('GET', Module.wasmUrl, true);
+    xhr.responseType = 'arraybuffer';
+    xhr.onload = function() {
+        if (typeof Wasm == 'undefined' && WebAssembly.compile) {
+            WebAssembly.compile(xhr.response).then(function(module) {
```

Comment:
`WebAssembly.compile` was defined to return a `Promise` from day 1; are you confusing this with `Wasm.instantiateModule`?